### PR TITLE
handle subdocs

### PR DESCRIPTION
**Pull request by IslandRhythms**

### CODE REVIEW

File: frontend/src/models/models.html [hunk 1]
```diff
@@ -28,7 +28,7 @@ <h1>Models</h1>
           <th v-for="path in schemaPaths">
             {{path.path}}
             <span class="path-type">
-              ({{(path.instance || 'unknown').toLowerCase()}})
+              ({{(path.instance ? path.instance : isSubdoc(path) ? 'subdocument' : 'unknown').toLowerCase()}})
```
Comment (vkarpov15) [inline frontend/src/models/models.html:31]: Shouldn't put this much logic into a template. Ideally you would move all this into a method. But I'll suggest an alternative fix for this particular issue.

File: frontend/src/models/models.js [hunk 2]
```diff
@@ -67,6 +68,16 @@ module.exports = app => app.component('models', {
       
       await this.getDocuments();
     },
+    isSubdoc(schemaPath) {
+      console.log('what is schemaPath in isSubdoc', schemaPath);
+      if (typeof schemaPath.options?.type?.tree === 'undefined') {
```
Comment (vkarpov15) [inline frontend/src/models/models.js:73]: So the issue here is that, for subdocuments, `schemaPath` is "corrupted". The issue comes down to this commit in Mongoose: https://github.com/Automattic/mongoose/commit/6399f3ca9084776a3d9688811e22f4a301452bc5 , which I will likely try to undo.

But I think the correct fix for this issue is to make sure that we always have `instance` for a schema path in `getDocuments()`. Something like `schemaPaths = {}; for (const path of Object.keys(Model.schema.paths)) { schemaPaths[path] = { instance: Model.schema.paths[path] } }`

**Review by vkarpov15 — CHANGES_REQUESTED** [review]

File: frontend/src/models/models.js (hunk 2 above)
Comment (IslandRhythms) [reply frontend/src/models/models.js]: I made your change but not exactly sure how this helps when you patch the issue. The way you have the frontend set up I'm  now having to add an extra`.instance` to where `schemaPath` is accessed or passed but doesn't seem to resolve the subdoc issue.

File: backend/actions/Model/getDocuments.js [hunk 3]
```diff
@@ -50,7 +50,10 @@ module.exports = ({ db }) => async function getDocuments(params) {
     sort({ _id: -1 });
 
 
-  const schemaPaths = Model.schema.paths;
+  let schemaPaths = {};
+  for (const path of Object.keys(Model.schema.paths)) {
+    schemaPaths[path] = { instance: Model.schema.paths[path]}
```
Comment (vkarpov15) [inline backend/actions/Model/getDocuments.js:55]: Make this `instance: Model.schema.paths[path].instance` so you don't need the repetitive `instance.instance` on the frontend

Comment (IslandRhythms) [reply backend/actions/Model/getDocuments.js]: I did that and the result was that there were other cases where we needed the rest of the object properties. I included my fix, let me know if you want it done differently.

**Review by vkarpov15 — APPROVED** [review]

**Pull request merged**